<?xml version="1.0"?>
<robot name="LWR" xmlns:xacro="http://www.ros.org/wiki/xacro">

<xacro:include filename="$(find lwr4p)/urdf/kuka_lwr_arm.xacro"/>
<xacro:include filename="$(find project_name_)/urdf/object.xacro"/>
<xacro:include filename="$(find project_name_)/urdf/frame_6DoF.xacro"/>


<property name="robot_name" value="lwr"/>

<!-- Inserts the Kuka here with a specified name and direction ... -->
<xacro:kuka_lwr_arm parent="base_link" name="${robot_name}" right="1" tool_name=""/>

<link name="world" />

<link name="base_link" />
<joint name="world_base_joint" type="fixed">
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <parent link="world"/>
    <child link="base_link"/>
</joint>

<link name="ee_link" />
<joint name="robot_ee_joint" type="fixed">
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <parent link="${robot_name}_arm_7_link"/>
    <child link="ee_link"/>
</joint>

<!-- ================================================= -->

<joint name="robot_object_link_joint" type="fixed">
    <origin xyz="0 0 0.04" rpy="0 0 0"/>
    <parent link="ee_link"/>
    <child link="robot_object_link"/>
</joint>

<link name="robot_object_link">

  <visual>
    <origin xyz="0 0 0" rpy="0 0 0" />
    <geometry>
      <cylinder radius="0.03" length="0.08" />
    </geometry>
    <material name="">
      <color rgba="0.65 0.65 0.65 1.0"/>
    </material>
  </visual>

</link>


<!-- ============================================================= -->

<xacro:object name="" parent="robot_object_link">
  <origin xyz="-0.1 0 0.04" rpy="0 0 0" />
</xacro:object>

<!-- left handle frame -->
<joint name="object_left_handle_frame_joint" type="fixed">
    <origin xyz="0 0 0" rpy="0 1.57 1.57"/>
    <parent link="left_handle_handle"/>
    <child link="left_handle_frame"/>
</joint>

<link name="left_handle_frame"/>

<!-- right handle frame -->
<joint name="object_right_handle_frame_joint" type="fixed">
    <origin xyz="0 0 0" rpy="0 1.57 1.57"/>
    <parent link="right_handle_handle"/>
    <child link="right_handle_frame"/>
</joint>

<link name="right_handle_frame"/>

<xacro:frame_6DoF name="frame" parent="world" />

</robot>
